src/Lambda.java

```diff
@@ -0,0 +1,175 @@
+        logger.log("Processing Bucket: " + bucketName); 
+ 
+        ObjectListing files = s3Client.listObjects(bucketName); 
+        List<KeyVersion> filesProcessed = new ArrayList<DeleteObjectsRequest.KeyVersion>(); 
+ 
+        for (Iterator<?> iterator = files.getObjectSummaries().iterator(); iterator.hasNext(); ) { 
+            S3ObjectSummary summary = (S3ObjectSummary) iterator.next(); 
+            logger.log("Reading Object: " + summary.getKey()); 
+ 
+            String trackingNumber = summary.getKey().split("--")[0]; 
+            Pair<Long, String> lastKnownStatus = latestStatusForTrackingNumber.get(trackingNumber); 
+ 
+            // Check if this shipment has already been delivered, skip this file 
+            if (lastKnownStatus != null && "DELIVERED".equals(lastKnownStatus.getRight())) { 
+                continue; 
+            } 
+ 
+            String fileContents = s3Client.getObjectAsString(bucketName, summary.getKey()); 
+ 
+            if (!isValidFile(fileContents)) { 
+                logger.log(String.format("Skipping invalid file %s", summary.getKey())); 
+                continue; 
+            } 
+             
+            if (!fileContents.contains("\n")) { 
+                 
+            } 
+            String[] lines = fileContents.split("\n"); 
+            String line1 = lines[0]; 
+            String line2 = lines[1]; 
+ 
+            String status = line1.split(":")[1]; 
+            Long timeStamp = Long.parseLong(line2.split(":")[1]); 
+ 
+ 
+            if (null == lastKnownStatus || lastKnownStatus.getLeft() < timeStamp) { 
+                lastKnownStatus = new MutablePair<Long, String>(timeStamp, status); 
+                latestStatusForTrackingNumber.put(trackingNumber, lastKnownStatus); 
+            } 
+ 
+            //Add to list of processed files 
+            filesProcessed.add(new KeyVersion(summary.getKey())); 
+            logger.log("logging Contents of the file" + fileContents); 
+        } 
+        return filesProcessed; 
+    } 
+     
+ 
+    private void deleteProcessedFiles(Map<String, List<KeyVersion>> filesToDelete) { 
+      final AmazonS3 s3Client = EventHandler.getS3Client(); 
+      for (Entry<String, List<KeyVersion>> entry : filesToDelete.entrySet()) { 
+          final DeleteObjectsRequest deleteRequest = new DeleteObjectsRequest(entry.getKey()).withKeys(entry.getValue()).withQuiet(false); 
+          s3Client.deleteObjects(deleteRequest); 
+      } 
+    } 
+     
+    private boolean isValidFile(String fileContents) { 
+        if (!fileContents.contains("\n")) { 
+            return false; 
+        } 
+        String[] lines = fileContents.split("\n"); 
+        for (String l: lines) { 
+            if (!l.contains(":")) { 
+                return false; 
+            } 
+        } 
+        return true; 
+    } 
+     
+    public static AmazonS3 getS3Client() { 
+        return AmazonS3ClientBuilder.standard().withRegion(Regions.DEFAULT_REGION).build(); 
```

Comment:
Recommendation generated by Amazon CodeGuru-Reviewer
This code is written so that the client cannot be reused across invocations of the Lambda function.
To improve the performance of the Lambda function, consider using static initialization/constructor, global/static variables and singletons. It allows to keep alive and reuse HTTP connections that were established during a previous invocation.
Learn more about [best practices for working with AWS Lambda functions](https://docs.aws.amazon.com/lambda/latest/dg/best-practices.html).